pilot/pkg/security/authz/builder/builder.go

```diff
@@ -47,29 +53,50 @@ func NewBuilder(serviceInstance *model.ServiceInstance, policies *model.Authoriz
 	serviceName := serviceInstance.Service.Attributes.Name
 	serviceNamespace := serviceInstance.Service.Attributes.Namespace
 	serviceHostname := string(serviceInstance.Service.Hostname)
-	if !policies.IsV1RbacEnabled(serviceHostname, serviceNamespace) {
-		rbacLog.Debugf("RBAC disabled for service %s", serviceHostname)
-		return nil
-	}
-
 	serviceMetadata, err := authz_model.NewServiceMetadata(serviceName, serviceNamespace, serviceInstance)
 	if err != nil {
 		rbacLog.Errorf("failed to create ServiceMetadata for %s: %s", serviceName, err)
 		return nil
 	}
 
 	isGlobalPermissiveEnabled := policies.IsGlobalPermissiveEnabled()
-	generator := v1.NewGenerator(serviceMetadata, policies, isGlobalPermissiveEnabled)
 
-	return &Builder{
+	builder := &Builder{
 		isXDSMarshalingToAnyEnabled: isXDSMarshalingToAnyEnabled,
-		generator:                   generator,
 	}
+
+	if policies.IsRBACEnabled(serviceHostname, serviceNamespace) {
+		builder.v1alpha1Generator = v1alpha1.NewGenerator(serviceMetadata, policies, isGlobalPermissiveEnabled)
+	} else {
+		rbacLog.Debugf("v1alpha1 RBAC policy disabled for service %s", serviceHostname)
+	}
+
+	// TODO: support policy in root namespace.
+	matchedPolicies := policies.ListAuthorizationPolicies(serviceNamespace, serviceMetadata.Labels)
```

Comment:
Should we still use ServiceInstance for workload selector matching? It seems that sidecar is matching workload selector with labels on a Proxy node (see https://github.com/istio/istio/blob/78bd9364c0cbaa132da2a1b4753d6668136a4859/pilot/pkg/model/context.go#L239).

Reply:
I'm not clear how is SidecarScope and the link you provided is related here? I think the right reference should be here: https://github.com/istio/istio/blob/9bb9df2cbcc3e8e54196273ce32cf9038c4ee491/pilot/pkg/networking/core/v1alpha3/gateway.go#L57 for Gateway?

The label is from the ServiceInstance which is actually the same pod labels: https://github.com/istio/istio/blob/9153840e8149ac9f5ed0e7eb7ac5590022ac2a95/pilot/pkg/serviceregistry/kube/controller/controller.go#L482, but I think it's a good point to make our implementation consistent with others.